test/test_sbol2_sbol3_direct.py

```diff
@@ -14,140 +14,84 @@
 
 class TestDirectSBOL2SBOL3Conversion(unittest.TestCase):
 
-    # TODO: turn on validation
-
-    # J23101.xml is not SBOL compliant. Leaving conversions involving it for after the compliant converter is done
-    '''
-    def test_3to2_conversion(self):
-        """Test ability to convert a simple part from SBOL3 to SBOL2"""
-        # Load an SBOL3 document and check its contents
-        doc3 = sbol3.Document()
-        doc3.read(TEST_FILES / 'BBa_J23101_patched.nt')
-        # Convert to SBOL2 and check contents
-        doc2 = convert3to2(doc3, True)
-        #report = doc2.validate()
-        #self.assertEqual(len(report), 0, f'Validation failed: {report}')
-        with tempfile.TemporaryDirectory() as tmpdir:
-            tmp2 = Path(tmpdir) / 'doc2.xml'
-            doc2.write(tmp2)
-            self.assertFalse(file_diff(str(tmp2), str(TEST_FILES / 'BBa_J23101.xml')))
-            doc3_loop = convert2to3(doc2, use_native_converter=True)
-            self.assertEqual(len(doc3_loop.validate()), 0)
-            tmp3 = Path(tmpdir) / 'doc3_loop.nt'
-            doc3_loop.write(tmp3)
-            self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / 'BBa_J23101_patched.nt')))
-    '''
-
-    # J23101.xml is not SBOL compliant. Leaving conversions involving it for after the compliant converter is done
-    '''
-    def test_2to3_conversion(self):
+    def handle_2to3_conversion(self, test_filename, rubric_filename):
+        """Provides a re-usable test handler for converting SBOL3 to SBOL2 with different test files
+        test_filename: Name of an SBOL2 test file
+        rubric_filename: Name of an SBOL3 file with expected conversion
+    """
         """Test ability to convert a simple part from SBOL2 to SBOL3"""
+
         # Load an SBOL2 document and check its contents
         doc2 = sbol2.Document()
-        doc2.read(TEST_FILES / 'BBa_J23101.xml')
+        doc2.read(TEST_FILES / test_filename)
+
         # Convert to SBOL3 and check contents
         doc3 = convert2to3(doc2, use_native_converter=True)
         self.assertEqual(len(doc3.validate()), 0)
         with tempfile.TemporaryDirectory() as tmpdir:
             tmp3 = Path(tmpdir) / 'doc3.nt'
             doc3.write(tmp3)
-            self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / 'BBa_J23101_patched.nt')))
+            self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / rubric_filename)))
             doc2_loop = convert3to2(doc3, True)
-            # report = doc2.validate()
```

Comment:
Please don't delete these two commented-out validation lines, or the TODO for turning them back on. We need to make sure the SBOL2 documents do validate someday.

Reply:
I think I did some sloppy refactoring, cutting/pasting.  Anyway, https://github.com/SynBioDex/SBOL-utilities/pull/317/commits/bc085467746ce45e1e895c0321b80bd40fcebee7 now validates both the intermediate and the final round-tripped file generated by the converters.  

Also, added a few more comments just to make it a little more readable.